app/handlers/contact_infos_confirm.rb

```diff
@@ -15,6 +15,10 @@ def handle
     fatal_error(code: :no_contact_info_for_code) if params[:code].nil?
     contact_info = ContactInfo.where(confirmation_code: params[:code]).first
     fatal_error(code: :no_contact_info_for_code) if contact_info.nil?
+    user = contact_info.user
+    # the came_from url is only used when user is signing up
+    outputs[:stored_url] = user.came_from if user.is_temp?
```

Comment:
Actually, this alternate may be clearer (and doesn't require overriding the `registration_redirect_url` like I did above.

```ruby
handle_with(ContactInfosConfirm,
            complete: lambda {
              # If in the middle of registration, redirect to continue process
              user = @handler_result.outputs.contact_info.user
              if user.is_temp?
                redirect_to user.registration_redirect_url,
                            notice: 'Thanks for adding your email address.'
              else
                render :confirm, status: @handler_result.errors.any? ? 400 : 200
              end
            })
```